src/k8s/pkg/k8sd/app/hooks_join.go

```diff
@@ -16,6 +18,11 @@ import (
 func (a *App) onPostJoin(s *state.State, initConfig map[string]string) error {
 	snap := a.Snap()
 
+	joinClusterConfig, err := apiv1.JoinClusterConfigFromMap(initConfig)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal join config: %w", err)
+	}
```

Comment:
```suggestion
    if joinClusterConfig, err := apiv1.JoinClusterConfigFromMap(initConfig); err != nil {
	    return fmt.Errorf("failed to unmarshal join config: %w", err)
    }
```

Reply:
@eaudetcobello I think this is not valid in all the statements. This will scope the `joinClusterConfig` within the if statement.

Reply:
https://go.dev/play/p/OG3j0obkKCr

Reply:
Oh cool! Didn't know that. Thanks for the link!